### PR TITLE
More compact approximate histogram

### PR DESCRIPTION
**Pull request by navis**
current approx histogram supports two different binary format which would be selected automatically. compact format is preferred one but when it's does not meet required condition, sparse format would be used.
for sparse format, one element occupies 12 byte and it can be huge when the column is aggregating 100+ elements.

for compact format, it's not efficient when the cardinality is small. and also, it induces overhead on deserializing (should process values one by one or make map/list for values and sort them)

we've found 80% of segment is occupied by single approxHistogram column. wish to make this better.

---

basically the same with dense format but uses vint for count when count > 1. tag is appended to 0x01 rather than Long.MIN_VALUE to make vints efficient.

```
vint size (tagged with exact)
vint bin count
for each 8 values {
  float[] value[]
  byte marker (bitmap that true for count > 1) 
  vint count (tagged with approx, for count > 1)
}
optional for non-exact {
  float min
  float max
}
```

### CODE REVIEW

**Comment by fjy** [comment]
@navis is this backwards compatible?

File: processing/src/main/java/io/druid/segment/ByteBuffers.java [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+import java.nio.ByteBuffer;
+
+/**
+ */
+public class ByteBuffers
+{
+  private static final Unsafe UNSAFE;
+  private static final long ADDRESS_OFFSET;
+
+  static {
```
Comment (xvrl) [inline processing/src/main/java/io/druid/segment/ByteBuffers.java]: I'm not sure we should move this block into the main code until we actually use it outside of the benchmark, it could cause an unwanted exception when running on a vm that doesn't allow unsafe methods.

Comment (navis) [inline processing/src/main/java/io/druid/segment/ByteBuffers.java]: ok. I'll revert that.

**Comment by navis** [comment]
@fjy can be configured what to be used. but cannot read each other from binary format.

**Comment by navis** [comment]
seemed better to be separated from existing approxHistogram rather than adding boolean param. removing WIP tag anyway.
